advancedstats.py

```diff
@@ -0,0 +1,187 @@
+	return 100 * opp_points_allowed / poss
+
+def get_season_totals(data):
+	season_team_totals = data.groupby(['Season', 'TeamID_1']).agg({
+		'Score_1': {
+			'GP': 'count',
+			'Score_1': 'sum'
+		},
+		'Score_2': 'sum',
+		'FGM_1': 'sum',
+		'FGA_1': 'sum',
+		'FGM3_1': 'sum',
+		'FGA3_1': 'sum',
+		'FTM_1': 'sum',
+		'FTA_1': 'sum',
+		'OR_1': 'sum',
+		'DR_1': 'sum',
+		'Ast_1': 'sum',
+		'TO_1': 'sum',
+		'Stl_1': 'sum',
+		'Blk_1': 'sum',
+		'PF_1': 'sum',
+		'FGM_2': 'sum',
+		'FGA_2': 'sum',
+		'FGM3_2': 'sum',
+		'FGA3_2': 'sum',
+		'FTM_2': 'sum',
+		'FTA_2': 'sum',
+		'OR_2': 'sum',
+		'DR_2': 'sum',
+		'Ast_2': 'sum',
+		'TO_2': 'sum',
+		'Stl_2': 'sum',
+		'Blk_2': 'sum',
+		'PF_2': 'sum'
+	}).reset_index()
+	season_team_totals.columns = ['Season', 'TeamID_1', 'GP', 'Score_1', 'Score_2', 'FGM_1', 'FGA_1', 'FGM3_1',
+								  'FGA3_1', 'FTM_1', 'FTA_1', 'OR_1', 'DR_1', 'Ast_1', 'TO_1',
+								  'Stl_1', 'Blk_1', 'PF_1', 'FGM_2', 'FGA_2', 'FGM3_2', 'FGA3_2', 'FTM_2', 'FTA_2',
+								  'OR_2', 'DR_2', 'Ast_2',
+								  'TO_2', 'Stl_2', 'Blk_2', 'PF_2']
+	return season_team_totals
+
+def get_four_factors(tm_FGA, tm_FGM, tm_FGM3, tm_FTA, tm_FTM, tm_TO, tm_OR, opp_DR):
+
+	# Shooting the ball
+	efg = (tm_FGM + 0.5*tm_FGM3)/tm_FGA
+
+	# Taking care of the ball
+	turnover_rate = tm_TO / (tm_FGA + 0.44 * tm_FTA + tm_TO)
+
+	# Offensive rebounding
+	offensive_rebounding_percentage = tm_OR / (tm_OR + opp_DR)
+
+	# Free throw rate
+	free_throw_rate = tm_FTM / tm_FGA
+
+	return 0.4 * efg + 0.25 + turnover_rate + 0.2 * offensive_rebounding_percentage + 0.15 * free_throw_rate
+
+
+def get_advanced_statistics(data):
+	'''
+	Returns data in the format:
+	| Season | TeamID | Poss | OffRtg | DefRtg |
+
+	Note: Currently inefficient
+	:param data:
+	:return: advanced statistics for teams in a season
+	'''
+
+	# Calculate season totals for each team by season
+	season_team_totals = get_season_totals(data)
+
+	# Calculate advanced statistics
+	# Calculate possessions for each team
+	season_team_totals['Total_Poss'] = season_team_totals.apply(lambda season_team_total: get_poss(
+		season_team_total['FGA_1'],
+		season_team_total['FGM_1'],
+		season_team_total['FTA_1'],
+		season_team_total['OR_1'],
+		season_team_total['DR_1'],
+		season_team_total['TO_1'],
+		season_team_total['FGA_2'],
+		season_team_total['FGM_2'],
+		season_team_total['FTA_2'],
+		season_team_total['OR_2'],
+		season_team_total['DR_2'],
+		season_team_total['TO_2']
+	), axis=1)
+
+	# Calculate avg poss for a team
+	season_team_totals['Poss'] = season_team_totals.apply(lambda season_team_stat: season_team_stat['Total_Poss'] / season_team_stat['GP'], axis=1)
+
+	# Calculate offensive and defensive ratings
+	season_team_totals[['OffRtg', 'DefRtg']] = season_team_totals.apply(lambda season_team_stat: pd.Series({'OffRtg': get_ORtg(season_team_stat['Score_1'], season_team_stat['Total_Poss']), 'DefRtg': get_DRtg(season_team_stat['Score_2'], season_team_stat['Total_Poss'])}), axis=1)
```

Comment:
wouldn't it be easier to do this ?
```
season_team_totals['OffRtg'] = season_team_totals.apply(lambda row: 100*row.Score_1/row.Total_Poss, axis=1)
season_team_totals['DefRtg'] = season_team_totals.apply(lambda row: 100*row.Score_2/row.Total_Poss, axis=1)
``

Reply:
Just in case we need OffRtg and DefRtg for something else